crates/task-impls/src/consensus.rs

```diff
@@ -537,6 +537,15 @@ where
                         .await;
                 }
             });
+            let consensus = self.consensus.read().await;
+            consensus
+                .metrics
+                .current_view
+                .set(usize::try_from(self.cur_view.get_u64()).unwrap());
+            consensus.metrics.number_of_views_since_last_decide.set(
+                usize::try_from(self.cur_view.get_u64()).unwrap()
+                    - usize::try_from(consensus.last_decided_view.get_u64()).unwrap(),
+            );
```

Comment:
From the code comment, looks like the `last_decided_view` of the `Consensus` struct is indeed the "last view (that) had a successful decide event", so I think this is the correct way to calculate how many views since a decide event.

Is there another data that we should use instead to get this info? Or were you actually suggesting moving this metric update to somewhere else so that we can get different numbers rather than `2` most of the time? Maybe I misunderstood your comment. :)

Reply:
Oh I didn't realize that is what `last_decided_view` was storing.  If it stores what you said then we can ignore my comment. :)

Reply:
This value is 2 most of the time indeed. We also have a metric on the number of views timed out. iiuc, I'll leave it as it is now. But definitely let me know if you suggest adding a new metric to record something different :)

Reply:
Sounds good!